8.9/apache-buster/Dockerfile

```diff
@@ -57,17 +56,23 @@ RUN { \
 		echo 'opcache.fast_shutdown=1'; \
 	} > /usr/local/etc/php/conf.d/opcache-recommended.ini
 
-WORKDIR /var/www/html
-
 # https://www.drupal.org/node/3060/release
 ENV DRUPAL_VERSION 8.9.2
-ENV DRUPAL_MD5 f3f64fbef97831efd1c943f392975f45
 
+# https://github.com/drupal/drupal/blob/9.0.1/composer.lock#L4052-L4053
+COPY --from=composer:1.10 /usr/bin/composer /usr/local/bin/
 RUN set -eux; \
-	curl -fSL "https://ftp.drupal.org/files/projects/drupal-${DRUPAL_VERSION}.tar.gz" -o drupal.tar.gz; \
-	echo "${DRUPAL_MD5} *drupal.tar.gz" | md5sum -c -; \
-	tar -xz --strip-components=1 -f drupal.tar.gz; \
-	rm drupal.tar.gz; \
-	chown -R www-data:www-data sites modules themes
+	export COMPOSER_HOME="$(mktemp -d)"; \
```

Comment:
Going forward, composer is an essential tool to Drupal, it is not just for installing Drupal for the first time. e.g. Drupal also uses composer to do patch management, Drupal also needs composer to install new modules and themes. I highly recommend that we don't put composer in a temp directory, and delete the executable after we build this image.

Reply:
Besides that, other things LGTM

Reply:
@skyred This raises an important question, I think: should Composer always be included the base Drupal image? This may not always be desirable.

Instead, I wonder if the Drupal image should focus primarily on providing all the dependencies necessary to *run* Drupal 9 in production. And then, downstream users of it can decide whether or not they want to extend it and install Composer in images used for development and/production for their own use-cases.

I like the idea of only installing Composer locally for development purposes, using it to build a production image, and then deploying that as an artifact (eg: using [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/)). If Composer is installed in the base Drupal image, it requires an extra step to remove it.

There could be a good discussion of the pros/cons of including Composer, but I'm not sure that needs to happen here. Perhaps we should leave this as-is (not include Composer in the official Drupal image), and open a new issue to discuss different strategies for including it in development/production? (Similar to the volume discussion #3 )

I'm also curious how this is being handled in other official Docker images for projects that use Composer - do they keep it in the image, or do they remove it like @yosifkit did here? Maybe this is already an established pattern?

Curious to hear your and others' thoughts on this!

Reply:
I'm confused, these Dockerfiles `COPY` in `composer` from the composer image and do not remove it.  The only thing deleted is the cache that composer creates in `COMPOSER_HOME`.

As for other official-images that include `composer`, I don't think there are any that use or have `composer` except the `composer` image.

Reply:
Ah I'm sorry I misunderstood - looking at the code now it does not delete `composer`. @skyred maybe you were suggesting we *should* be deleting `composer`?

I do not feel strongly either way - downstream images can always choose to remove `composer` if they want to.

I'm curious to hear how others deal with `composer` in production, in either case, but that doesn't have to block this from merging! :-)

Reply:
Sorry for the confusion. I thought we were deleting composer excusable. Deleting cache makes sense. 

I like keep composer for two reasons: 1. many production sites rely on composer for patch management 2. [Automated updates initiative](https://www.drupal.org/docs/8/update/automatic-updates) would need composer.

LGTM